papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/controller/ProjectsController.java

```diff
@@ -98,4 +103,53 @@ private Project getValidProject(Long id) {
 
         return ResponseEntity.ok(sprints);
     }
+
+    @Override
+    public ResponseEntity<UserAvailabilityDto> getUserAvailability(Long projectId, Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+
+        if (!isSprintInProject(sprint, project)) {
+            throw new ForbiddenAccessException();
+        }
+
+        Availability availability = availabilityService
+                .findBySprintIdAndUserId(sprintId, userContext.getUserIdn())
+                .orElseThrow(ResourceNotFoundException::new);
+
+        return ResponseEntity.ok(availabilityMapper.mapToApi(availability));
+    }
+
+    @Override
+    public ResponseEntity<UserAvailabilityDto> updateUserAvailability(@Valid UserAvailabilityDto body, Long projectId,
+            Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+        if (!isSprintInProject(sprint, project) || !isAvailabilityDeclarationAllowed(sprint)) {
+            throw new ForbiddenAccessException();
+        }
+
+        Availability availability = availabilityService
+                .findBySprintIdAndUserId(sprintId, userContext.getUserIdn()).orElseGet(Availability::new);
+
+        availability.setNotes(body.getNotes());
+        availability.setPastSprintRemainingTime(Duration.ofMinutes(body.getTimeRemaining()));
+        availability.setTimeAvailable(Duration.ofMinutes(body.getTimeAvailable()));
+        availabilityService.save(availability);
```

Comment:
Nie wiem na ile akceptujesz ustalenia z #19 o zarządzaniu życiem obiektów systemowych, ale tylko o tym wspominam.

Reply:
MIŁEJ LEKTURY (tam sobie musisz porozwijać komentarze jakby co)

Reply:
Mam nadzieję, że nie proponujesz wizarda
Czy zrobienie w serwisie metody update wywoływanej z kontrolera, która będzie miała taką samą logikę jak powyżej, byłoby dla was okej?

Reply:
Myślę, że wszyscy proponują wizarda

Reply:
No okej, ale nie rozumiem dlaczego. Argument że kontroler nie powinien zarządzać bezpośrednią persystencją nawet do mnie trafia, ale dlaczego w takim razie nie może tego robić serwis (który jest w core)

Reply:
@szbrnl z mojej strony, jeśli uważasz że warstwy prezentacji i persystencji są ze sobą wystarczająco luźno powiązane i decyzja jest świadoma to nie mam nic do tego

Reply:
@kjarosh mógłbyś sprawdzić obecną wersję?